lib/screens_web/controllers/v2/screen_api_controller.ex

```diff
@@ -84,19 +84,29 @@ defmodule ScreensWeb.V2.ScreenApiController do
 
         response =
           screen_id
-          |> screen_response(variant, run_all_variants?: true, update_visible_alerts?: true)
+          |> screen_response(variant, screen,
+            run_all_variants?: true,
+            update_visible_alerts?: true
+          )
           |> put_extra_fields(screen_id, screen)
 
         json(conn, response)
     end
   end
 
-  defp screen_response(screen_id, "all", opts) do
+  defp screen_response(screen_id, "all", _, opts) do
```

Comment:
Minor quibble: The order of arguments being `screen_id, variant, screen` feels off to me — I'd expect `screen_id, screen, variant`, so the "parts of the screen" are together.